File: app/src/main/java/io/renderapps/balizinha/AppController.java
package io.renderapps.balizinha;

import android.app.Application;

import com.crashlytics.android.Crashlytics;
import com.google.firebase.database.FirebaseDatabase;

import io.fabric.sdk.android.Fabric;

/**
 * Created by joel
 * on 12/12/17.
 */
public class AppController extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        FirebaseDatabase.getInstance().setPersistenceEnabled(true);

        final Fabric fabric = new Fabric.Builder(this)
                .kits(new Crashlytics())
                .debuggable(true)
                .build();
        Fabric.with(fabric);
    }

    // KitKat support
//    @Override
//    protected void attachBaseContext(Context base) {
//        super.attachBaseContext(base);
//        MultiDex.install(this);
//    }
}

File: app/src/main/java/io/renderapps/balizinha/AppController.kt
package io.renderapps.balizinha

import android.app.Application
import com.crashlytics.android.Crashlytics
import io.fabric.sdk.android.Fabric

import io.renderapps.balizinha.data.prefs.SharedPrefsHelper
import io.renderapps.balizinha.data.DataManager

class AppController : Application() {

    private lateinit var dataManager: DataManager

    override fun onCreate() {
        super.onCreate()

//        if (LeakCanary.isInAnalyzerProcess(this)) {
//            // This process is dedicated to LeakCanary for heap analysis.
//            // You should not init your app in this process.
//            return
//        }
//
//        LeakCanary.install(this)


        val fabric = Fabric.Builder(this)
                .kits(Crashlytics())
                .debuggable(true)
                .build()
        Fabric.with(fabric)

        val sharedPrefsHelper = SharedPrefsHelper(applicationContext)
        dataManager = DataManager(sharedPrefsHelper)
    }

    fun getDataManager(): DataManager {
        return dataManager
    }
}